commands/general/help.js

```diff
@@ -1,55 +1,98 @@
 const Command = require(`${process.cwd()}/base/Command.js`);
-
+const {MessageEmbed} = require("discord.js");
+const perpage = 10;
+/*
+  The HELP command is used to display every command's name and description
+  to the user, so that he may see what commands are available. The help
+  command is also filtered by level, so if a user does not have access to
+  a command, it is not shown to them. If a command name is given with the
+  help command, its extended help is shown.
+*/
 class Help extends Command {
   constructor(client) {
     super(client, {
       name: "help",
-      description: "Displays the commands for your level.",
-      usage: "help [command]",
+      description: "Get help on a command, command category, or a setting",
       category: "System",
-      extended: "This command will display all available commands for your permission level, with the additonal option of getting per command information when you run 'help <command name>'.",
-      hidden: true,
-      aliases: ["h", "halp", "commands"]
+      usage: "help <category/command/setting> [page-num]"
     });
   }
 
   async run(message, args, level) {
-    const settings = message.settings;
+
+    const embed = new MessageEmbed()
+      .setTimestamp()
+      .setColor(message.guild.me.roles.highest.color || 0x00AE86)
+      .setFooter(`Requested by ${message.author.tag}`, message.author.avatarURL()); 
+    // Preload MessageEmbed.
+  
+    // Here we sort out categories in case the user did not provide an argument.
+    
+    let currentCategory = "";
+    const sorted = this.client.commands.array().sort((p, c) => p.help.category > c.help.category ? 1 :  p.help.name > c.help.name && p.help.category === c.help.category ? 1 : -1 );
```

Comment:
There is no reason to make this an array tbh

Reply:
There is, sorted gets run into a .forEach()